src/wasm/wasm-type.cpp

```diff
@@ -349,25 +334,11 @@ bool Type::isNullable() const {
   }
 }
 
-size_t Type::size() const { return expand().size(); }
-
-const TypeList& Type::expand() const {
-  if (isBasic()) {
-    return basicTuples[id];
-  } else {
-    auto* typeDef = getDef(*this);
-    assert(typeDef->isTuple() && "can only expand tuple types");
-    return typeDef->tuple.types;
-  }
-}
-
 bool Type::operator<(const Type& other) const {
-  const TypeList& these = expand();
-  const TypeList& others = other.expand();
-  return std::lexicographical_compare(these.begin(),
-                                      these.end(),
-                                      others.begin(),
-                                      others.end(),
+  return std::lexicographical_compare((*this).begin(),
+                                      (*this).end(),
```

Comment:
I think these would be readable as `this->begin()` and `this->end()` or possibly just as `begin()` and `end()`.